be/src/exec/vectorized/analytor.cpp

```diff
@@ -666,6 +687,23 @@ int64_t Analytor::_find_first_not_equal(vectorized::Column* column, int64_t targ
     return end - 1;
 }
 
+void Analytor::_find_candidate_partition_ends() {
+    if (!_partition_statistics.is_high_cardinality()) {
+        return;
+    }
+
+    SCOPED_TIMER(_partition_search_timer);
+    for (size_t i = _found_partition_end + 1; i < _partition_columns[0]->size(); ++i) {
+        for (auto& column : _partition_columns) {
+            auto cmp = column->compare_at(i - 1, i, *column, 1);
```

Comment:
Why not further optimize this compare procedure into a column-wise way to eliminate the `compare_at` virtual function call?

For example, compare the first column, to get a `compare_vector`, in which `0` denotes not equal and `1` denotes equal. Then compare the second column for equal elements only.

The `build_tie_for_column` and `ColumnTieBuilder` could save some work for you.

Reply:
I try to use tie, and find that it is much more slower than the simple loop.